### PR TITLE
feat(result): new componet Result

### PR DESCRIPTION
**Pull request by kooriookami**
re #1137

Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/result/src/index.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      <slot v-if="$slots.icon" name="icon"></slot>
+      <i v-else :class="typeClass"></i>
+    </div>
+    <div v-if="title || $slots.title" class="el-result__title">
+      <slot v-if="$slots.title" name="title"></slot>
+      <p v-else>{{ title }}</p>
+    </div>
+    <div v-if="subTitle || $slots.subTitle" class="el-result__subtitle">
+      <slot v-if="$slots.subTitle" name="subTitle"></slot>
+      <p v-else>{{ subTitle }}</p>
```
Comment (JeremyWuuuuu) [inline packages/result/src/index.vue:13]: Please modify the syntax here.
```vue
<slot name="subTitle">
  <p> {{ subTitle }}</p>
</slot>
```

File: packages/result/src/index.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<template>
+  <div class="el-result">
+    <div class="el-result__icon">
+      <slot v-if="$slots.icon" name="icon"></slot>
+      <i v-else :class="typeClass"></i>
+    </div>
+    <div v-if="title || $slots.title" class="el-result__title">
+      <slot v-if="$slots.title" name="title"></slot>
+      <p v-else>{{ title }}</p>
+    </div>
```
Comment (JeremyWuuuuu) [inline packages/result/src/index.vue:10]: Same here.

File: packages/result/src/index.vue [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<template>
+  <div class="el-result">
+    <div class="el-result__icon">
+      <slot v-if="$slots.icon" name="icon"></slot>
+      <i v-else :class="typeClass"></i>
```
Comment (JeremyWuuuuu) [inline packages/result/src/index.vue:5]: Same here.

File: packages/result/src/index.vue [hunk 4]
```diff
@@ -22,16 +22,26 @@
...
+import IconError from './icon-error.vue'
+import IconWarning from './icon-warning.vue'
+import IconInfo from './icon-info.vue'
 
-const TypeMap: Indexable<string> = {
-  success: 'success',
-  info: 'info',
-  warning: 'warning',
-  error: 'error',
+const IconMAP: Indexable<string> = {
```
Comment (JeremyWuuuuu) [inline packages/result/src/index.vue:30]: use `ICON_MAP` or `iconMap`

**Review by jw-foss — APPROVED** [review]

**Pull request merged**